src/main/java/com/treasure/hunt/view/swing/ClassListCellRenderer.java

```diff
@@ -0,0 +1,74 @@
+package com.treasure.hunt.view.swing;
+
+import com.treasure.hunt.utils.Reflections;
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.swing.*;
+import javax.swing.border.EmptyBorder;
+import java.awt.*;
+
+public class ClassListCellRenderer extends JPanel implements ListCellRenderer<Class> {
+
+    @Getter
+    @Setter
+    int hoverIndex = -1;
+    @Getter
+    @Setter
+    Class otherGeneric = null;
+    private JLabel nameLabel = new JLabel();
+    private JLabel genericLabel = new JLabel();
+
+    @Override
+    public Component getListCellRendererComponent(JList<? extends Class> list, Class value, int index, boolean isSelected, boolean cellHasFocus) {
+        boolean isHovered = index == hoverIndex;
+        boolean isAvailable = Reflections.interfaceGenericsClass(value) == otherGeneric;
+
+        if (otherGeneric == null)
```

Comment:
like to see consistent code style and I would go for {} everywhere even if not needed